src/ui/view/share/data/dropDown/dropDown.tsx

```diff
@@ -0,0 +1,157 @@
+import type { FC } from 'react'
+import { useMemo, useCallback, useState } from 'react'
+import { Collapsible } from '@/ui/component/collapsible/collapsible'
+import { SearchBar } from '@/ui/component/searchbar/searchbar'
+import { isSubstringOf } from '@/util/util'
+import { Tag } from '@/ui/component/tag/tag'
+import { DynamicCheckbox } from '@/ui/view/dataverse/component/dynamicCheckbox/dynamicCheckbox'
+import { NoResultFound } from '@/ui/view/dataverse/component/noResultFound/noResultFound'
+import './dropDown.scss'
+
+type SelectionItemType = 'checkbox'
+
+export type DropDownProps = {
+  placeholder: string
+  value: string[]
+  options: string[]
+  searchPlaceholder: string
+  selectionType: SelectionItemType
+  onChange: (value: string[]) => void
+  maxSearchResults?: number
+}
+
+const {
+  dropDown: { dropDownMaxSearchResults, dropDownTagsDisplayLimit }
+} = APP_ENV
+
+type DropDownInputProps = Pick<DropDownProps, 'value' | 'placeholder'> & {
+  handleChange: (v: string) => () => void
```

Comment:
If `handleChange` is intended to be a callback function, it is recommended to make the following adjustments:

- Rename it to `onChange` to align with common naming conventions for callback functions.
- Update its definition to return `void` directly, instead of returning another function `() => void`.

Reply:
The parent component `DropDown` is already getting `onChange`, if I understand correctly your suggestion, this will me break [this lint rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md)

Reply:
So, you're telling me that you're twisting the return type to allow the call to be made in the props while respecting `jsx-no-bind rule`? Under no circumstances. Component properties are *public* interfaces that convey meaning and suggest behavior. Specifying a property as a callback that returns a function is nonsensical. The binding should be done in the core of the component, using the most appropriate approach (e.g. `useCallback` or whatever).

Reply:
I've had previous comments about this practice, your suggestion looks more convenient and follows the pattern I've always followed ☺️

Reply:
Ah, I see. Good news, thanks so much for your consideration.